src/hera/workflows/script.py

```diff
@@ -555,8 +555,11 @@ def _extract_return_annotation_output(source: Callable) -> List:
     if get_workflow_annotation(return_annotation):
         output.append(annotation_args)
```

Comment:
This is a good opportunity to clean up the function for correctness and better readability. Also `annotation_args` is only needed if the origin_type is tuple, so we can move that

```suggestion
    if param_or_artifact := get_workflow_annotation(return_annotation):
        output.append(param_or_artifact)
```

Reply:
This change breaks the tests. I tried to fix it, but seems the original type (annotation_args[0]) is needed.

Following lines are breaking 😢 
https://github.com/argoproj-labs/hera/blob/2eec4c282557823c661dae6b410b470d13a90be5/src/hera/workflows/_runner/script_annotations_util.py#L241-L254